support-frontend/assets/helpers/forms/paymentIntegrations/amazonPay/setup.ts

```diff
@@ -1,23 +1,25 @@
-import type { Dispatch } from 'redux';
 import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
 import { isProd } from 'helpers/urls/url';
 import { logException } from 'helpers/utilities/logger';
-import {
-	setAmazonPayLoginObject,
-	setAmazonPayPaymentsObject,
-} from 'pages/contributions-landing/contributionsLandingActions';
+import type { AmazonLoginObject, AmazonPaymentsObject } from './types';
 
 // ---- Setup ---- //
 
 export const setupAmazonPay = (
 	countryGroupId: CountryGroupId,
-	dispatch: Dispatch,
+	onAmazonLoginReady: (loginObject: AmazonLoginObject) => void,
+	onAmazonPaymentsReady: (paymentsObject: AmazonPaymentsObject) => void,
```

Comment:
The main change in this file is that the setup functions now take specific callbacks to set the objects instead of the generic dispatch function